apps/events/models.py

```diff
@@ -304,65 +307,83 @@ class Rule(models.Model):
         _("utsettelse"), help_text=_("utsettelse oppgis i timer"), default=0
     )
 
-    def get_offset_time(self, time):
+    def get_offset_time(self, time: timezone.datetime):
         if type(time) is not datetime:
             raise TypeError("time must be a datetime, not %s" % type(time))
         else:
             return time + timedelta(hours=self.offset)
 
-    def satisfied(self, user):
-        """ Checks if a user satisfies the rules """
+    def satisfies_constraint(self, user: User) -> bool:
         return True
 
-    def __str__(self):
-        return "Rule"
-
-    class Meta:
-        permissions = (("view_rule", "View Rule"),)
-        default_permissions = ("add", "change", "delete")
-
-
-class FieldOfStudyRule(Rule):
-    field_of_study = models.SmallIntegerField(
-        _("studieretning"), choices=FieldOfStudyType.ALL_CHOICES
-    )
-
-    def satisfied(self, user, registration_start):
+    def satisfied(self, user: User, registration_start: timezone.datetime):
         """ Override method """
 
-        # If the user has the same FOS as this rule
-        if self.field_of_study == user.field_of_study:
+        if self.satisfies_constraint(user):
             offset_datetime = self.get_offset_time(registration_start)
             # If the offset is in the past, it means you can attend even with the offset
             if offset_datetime < timezone.now():
-                return {"status": True, "message": None, "status_code": 210}
+                return {
+                    "status": True,
+                    "message": None,
+                    "status_code": self.success_code,
+                }
             # If there is no offset, the signup just hasn't started yet
             elif self.offset == 0:
                 return {
                     "status": False,
                     "message": _("Påmeldingen har ikke åpnet enda."),
-                    "status_code": 402,
+                    "status_code": self.not_opened_code,
                 }
             # In the last case there is a delayed signup
             else:
                 return {
                     "status": False,
-                    "message": _("Din studieretning har utsatt påmelding."),
+                    "message": self.delayed_singup_message,
                     "offset": offset_datetime,
-                    "status_code": 420,
+                    "status_code": self.delayed_signup_code,
                 }
         return {
             "status": False,
-            "message": _(
-                "Din studieretning er en annen enn de som har tilgang til dette arrangementet."
-            ),
-            "status_code": 410,
+            "message": self.not_satisfied_message,
+            "status_code": self.not_satisfied_code,
         }
 
+    def __str__(self):
+        return "Rule"
+
+    class Meta:
+        permissions = (("view_rule", "View Rule"),)
+        default_permissions = ("add", "change", "delete")
```

Comment:
By having it as an abstract model we loose all ability to handle Rules as their their own queryset. While it may not be needed now, its a not straight forward to get that functionality back, so I don't quite see the reason to remove it not.

Reply:
I do not see any reason for us to query over `Rules`, except for some usage-stats?
Could we at least add a comment either here, explaining that this isn't abstract to allow that specific usecase?

Reply:
I just don't think it makes that much of a difference in this case :shrug: